File: src/main/java/com/github/michael72/pumlsrv/ConverterResult.java
package com.github.michael72.pumlsrv;

public class ConverterResult {
  public ConverterResult(final byte[] bytes, final String description, final String image_type, final boolean isError) {
    this.bytes = bytes;
    this.description = description;
    this.image_type = image_type;
    this.isError = isError;
  }

  public final byte[] bytes;
  public final String description;
  public final String image_type;
  public final boolean isError;
}

File: src/main/kotlin/com/github/michael72/pumlsrv/ConverterResult.kt
package com.github.michael72.pumlsrv

data class ConverterResult(
    val bytes: ByteArray,
    val description: String,
    val imageType: String,
    val isError: Boolean
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ConverterResult

        if (!bytes.contentEquals(other.bytes)) return false
        if (description != other.description) return false
        if (imageType != other.imageType) return false
        if (isError != other.isError) return false

        return true
    }

    override fun hashCode(): Int {
        var result = bytes.contentHashCode()
        result = 31 * result + description.hashCode()
        result = 31 * result + imageType.hashCode()
        result = 31 * result + isError.hashCode()
        return result
    }
}